packages/console-types/src/types/services.ts

```diff
@@ -450,6 +450,26 @@ export const serviceSecret = {
   additionalProperties: false,
 } as const
 
+
+export const serviceAccount = {
+  type: 'object',
+  properties: {
+    name: serviceAccountName,
+    deleted: { type: 'boolean' },
+  },
+  additionalProperties: false,
+  required: ['name'],
+} as const
+
+export const serviceAccounts = {
+  type: 'object',
+  patternProperties: {
+    [serviceAccountName.pattern]: serviceAccount,
+  },
```

Comment:
We should use the additionalProperties. It is not relevant the pattern of the key

```suggestion
  additionalProperties: serviceAccount,
```

Reply:
I've copied the same configuration of configmaps and secrets, both of them have restrictions on the name of the resource on the kubernetes side and use this configuration for the key